### PR TITLE
[BUGFIX] Translations of nested content elements

### PR DESCRIPTION
**Pull request by maxmishyn**
It fixes the bug which makes impossible to render content of CE with the IRRE elements like BootstrapPackage's Icons list, Accordion and similar content elements with multiply subitems when placing them inside of the translated instance of the container. In the translated version container was able to render only part of the child content element stored in the tt_content. All records from the foreign tables (like accordion items) were skipped in the translated version.

### CODE REVIEW

**Comment by achimfritz** [comment]
thanks for fixing, i will provide a test for this case

**Comment by lukasniestroj** [comment]
Hi @achimfritz,

This "fix" breaks translations inside the property "renderedContent". 
I have a simple container and 2 languages configured. Also its TYPO3 10.4.14 and container:1.3.0

Currently only the tt_content of the default translation is rendered inside renderedContent. 
If I remove this "fix" everything works again. (I see the translated content)
I do not understand the initial intention. Should renderedContent always be the result of "l18n_parent" (if set)? 

If I take a look at the whole data object I see the correct language record. My current workaround is to render the content again inside the template...

**Comment by maxmishyn** [comment]
I just double checked in 4 languages environment. It looks ok both in fully translated and mixed modes.
The original intention was to fix translation issue of "nested" content elements (accordions, icon groups and etc with IRRE). Since the renderedContent was the translated record already (without the original language), typo3 didn't even try to translate children of such records. 
Maybe post your records here as @achimfritz suggested to review your case.

**Comment by Narfsn** [comment]
@achimfritz I had to dig into this for a bit because i stumbled upon the issue in a TYPO3 Site which uses fallbackType=free as a setting. 

What i found so far: The bugfix works with "strict" and "fallback" but the issue aka IRRE Elements not being shown is also only the case if the Site is configured as "fallback". Both "free" and "fallback" worked just fine without that line.

It seems TYPO3\CMS\Frontend\ContentObject\ContentObjectRenderer::render just accepts the uid and renders the default language. Using "strict" or "fallback" works, because the render applies the overlays.

TLDR:
- With the Bugfix "free" doesn't work as expected (Translated elements are in the default language)
- Without the Bugfix "fallback" IRRE Elements don't work at all. 

I tested something like this which seems to fix the problem in all 3 fallbackType's but i'm not sure if there's some caveat:

```
$languageAspect = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Context\Context::class)->getAspect('language');
$sys_language_uid = $languageAspect->getId();

$site = GeneralUtility::makeInstance(\TYPO3\CMS\Core\Site\SiteFinder::class)->getSiteByPageId($GLOBALS['TSFE']->id);

$fallbackType = $site->getLanguageById($sys_language_uid)->getFallbackType();

$conf['source'] = $child['uid'];

if ($fallbackType == 'strict' || $fallbackType == 'fallback') {
    if ($child['l18n_parent'] > 0) {
        $conf['source'] = $child['l18n_parent'];
    }
}

if ($child['t3ver_oid'] > 0) {
    $conf['source'] = $child['t3ver_oid'];
}
```

**Comment by maxmishyn** [comment]
@Narfsn I tried your code in my usage scenario and can confirm to @achimfritz  that it works fine.
